src/webapp/src/characters/charactersSlice.ts

```diff
@@ -0,0 +1,63 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+interface StateData {
+  results: StateCharacter[];
+  error?: string | null;
+}
+
+export interface StateCharacter {
+  id: number;
+  name: string;
+  species: string;
+  image: string;
+}
+interface ApiKnownError {
+  error: string;
+}
+const initialState = {
+  results: [] as StateCharacter[],
+  error: null,
+} as StateData;
+
+export const getCharacters = createAsyncThunk<
+  StateCharacter[],
+  void,
+  { rejectValue: ApiKnownError }
+>('characters/getCharacters', async (_, { rejectWithValue }) => {
+  const response = await fetch("https://rickandmortyapi.com/api/character");
+
+  if (response.status === 404) {
+    return rejectWithValue((await response.json()) as ApiKnownError);
+  }
+  const { results } = await response.json();
+
+  return results.map(({ species, image, id, name }: StateCharacter) => ({
+    species,
+    image,
+    id,
+    name,
+  }));
```

Comment:
Reversing those two will make us catch more errors and avoid surprises. 
```suggestion
  if (response.ok) {
    const { results } = await response.json();
    return results.map(({ species, image, id, name }: StateCharacter) => ({
      species,
      image,
      id,
      name,
    }));
  }
  
  return rejectWithValue((await response.json()) as ApiKnownError);
```

Can you give this a shot please?

Reply:
It makes sense!! I like the approach. Checking if the response is successful to do stuff, and in any other case return the rejection, yes, beautiful.